Cython/Utility/CppConvert.pyx

```diff
@@ -136,34 +103,22 @@ cdef object {{cname}}(const cpp_list[X]& v):
 
 #################### set.from_py ####################
 
-cdef extern from *:
-    cdef cppclass set "std::{{maybe_unordered}}set" [T]:
-        void insert(T&) except +
-
 @cname("{{cname}}")
-cdef set[X] {{cname}}(object o) except *:
-    cdef set[X] s
+cdef Tp {{cname}}(object o) except *:
+    cdef Tp s
     for item in o:
-        s.insert(<X>item)
+        s.insert(<Tp.value_type>item)
     return s
 
 
 #################### set.to_py ####################
 
 cimport cython
 
-cdef extern from *:
-    cdef cppclass cpp_set "std::{{maybe_unordered}}set" [T]:
-        cppclass const_iterator:
-            T& operator*()
-            const_iterator operator++()
-            bint operator!=(const_iterator)
-        const_iterator begin()
-        const_iterator end()
-
 @cname("{{cname}}")
-cdef object {{cname}}(const cpp_set[X]& s):
-    return {v for v in s}
+cdef object {{cname}}(const Tp& s):
+    # casting away the const is a bit dubious, but helps Cython do the iteration
+    return {v for v in <Tp&>s}
```

Comment:
This casting away the const feels slightly dubious to me.

It's because Cython infers the iterator type from `begin`, I wonder if it should try `const_begin` first for const types, but that feels like a separate change.